proof-producer/bin/proof-producer/include/nil/proof-generator/circuit_writer.hpp

```diff
@@ -0,0 +1,93 @@
+//---------------------------------------------------------------------------//
+// Copyright (c) 2024 Daniil Kogtev <oclaw@nil.foundation>
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//---------------------------------------------------------------------------//
+
+#ifndef PROOF_GENERATOR_CIRCUIT_WRITER_HPP
+#define PROOF_GENERATOR_CIRCUIT_WRITER_HPP
+
+#include <ostream>
+#include <vector>
+
+#include <boost/assert.hpp>
+
+#include <nil/blueprint/blueprint/plonk/circuit.hpp>
+#include <nil/crypto3/marshalling/zk/types/plonk/constraint_system.hpp>
+#include <nil/crypto3/zk/snark/arithmetization/plonk/constraint_system.hpp>
+
+
+namespace nil {
+    namespace proof_generator {
+
+
+        template <typename Endianness, typename BlueprintField> 
+        class circuit_writer {
+                public:
+                    using ArithmetizationType = nil::crypto3::zk::snark::plonk_constraint_system<BlueprintField>;
+                    using TTypeBase = nil::marshalling::field_type<Endianness>;
+                    using Circuit = nil::blueprint::circuit<ArithmetizationType>;
```

Comment:
Let's use `nil::crypto3::zk::snark::plonk_constraint_system<BlueprintField>`

Reply:
It is already used under `ArithmetizationType`  alias

Reply:
i mean use the same type as in prover: https://github.com/NilFoundation/placeholder/blob/b640712c1f95dac05151431376a502b3613c08d9/proof-producer/bin/proof-producer/include/nil/proof-generator/prover.hpp#L158

Reply:
You mean rename ArithmetizationType --> ConstraintSystem?

Reply:
Oh, got it, will fix

Reply:
done